### PR TITLE
feat: Add JSON schema support for GitLab

### PR DESCRIPTION
**Pull request by mnorbury**
fixes: https://github.com/cloudquery/cloudquery/issues/16361

### CODE REVIEW

File: plugins/source/gitlab/client/spec.go [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
 type Spec struct {
-	Token       string `json:"access_token,omitempty"`
-	BaseURL     string `json:"base_url,omitempty"`
-	Concurrency int    `json:"concurrency,omitempty"`
+	// An access token for your GitLab server. Instructions on how to generate an access token here.
+	Token string `json:"access_token,omitempty" jsonschema:"required,minLength=1"`
+	//	URL for your self hosted GitLab server. Leave empty for GitLab SaaS. Not all tables are supported for GitLab SaaS.
+	BaseURL string `json:"base_url,omitempty"`
+	// A best effort maximum number of Go routines to use. Lower this number to reduce memory usage.
+	Concurrency int `json:"concurrency,omitempty" jsonschema:"default:10000"`
```
Comment (disq) [inline plugins/source/gitlab/client/spec.go:20]:
```suggestion
	Concurrency int `json:"concurrency,omitempty" jsonschema:"minimum=1,default=10000"`
```

File: plugins/source/gitlab/client/spec/gen/main.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	cqjsonschema "github.com/cloudquery/codegen/jsonschema"
+	"github.com/invopop/jsonschema"
+)
+
+func main() {
+	fmt.Println("Generating JSON schema for plugin spec")
+	cqjsonschema.GenerateIntoFile(new(client.Spec), path.Join(currDir(), "../..", "schema.json"),
+		cqjsonschema.WithAddGoComments("github.com/cloudquery/cloudquery/plugins/source/gitlab/client", path.Join(currDir(), "../..")),
+		func(r *jsonschema.Reflector) {
+			r.NullableFromType = false
```
Comment (disq) [inline plugins/source/gitlab/client/spec/gen/main.go:19]: No need for this

File: plugins/source/gitlab/client/spec_test.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+			Err:  false,
+		},
+		{
+			Name: "concurrency",
+			Spec: `{"access_token":"token","concurrency":100}`,
+			Err:  false,
+		},
+		{
+			Name: "invalid field",
+			Spec: `{"unknown_field":"value"}`,
```
Comment (disq) [inline plugins/source/gitlab/client/spec_test.go:38]: Here we're not sure if it's failing from access_token or invalid field. Should add a valid config + unknown field, as in:

```suggestion
			Spec: `{"access_token":"token", "unknown_field":"value"}`,
```

Comment (mnorbury) [reply plugins/source/gitlab/client/spec_test.go]: Good point - committed.

**Review by disq — APPROVED** [review]

**Pull request merged**